semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/package.scala

```diff
@@ -4,76 +4,82 @@ import gremlin.scala.GremlinScala
+  class ArrayAccessExt(val arrayAccess: ArrayAccess) extends AnyVal {
+    def array: nodes.Expression = arrayAccess.argument(1)
+    def subscripts: NodeSteps[nodes.Identifier] = arrayAccess.argument(2).ast.isIdentifier
   }
 
-  class ArrayAccessTrav(val wrapped: Steps[nodes.Call]) extends AnyVal {
-    private def raw: GremlinScala[nodes.Call] = wrapped.raw
+  class ArrayAccessTrav(val wrapped: Steps[ArrayAccess]) extends AnyVal {
+    private def raw: GremlinScala[ArrayAccess] = wrapped.raw
     def array: NodeSteps[nodes.Expression] = wrapped.map(_.array)
     def subscripts: NodeSteps[nodes.Identifier] = wrapped.flatMap(_.subscripts)
   }
 
-  class AssignmentExt(val call: nodes.Call) extends AnyVal {
-    def target: nodes.Expression = call.argument(1)
-    def source: nodes.Expression = call.argument(2)
+  class AssignmentExt(val assignment: Assignment) extends AnyVal {
+    def target: nodes.Expression = assignment.argument(1)
+    def source: nodes.Expression = assignment.argument(2)
   }
 
-  class AssignmentTrav(val wrapped: Steps[nodes.Call]) extends AnyVal {
-    private def raw: GremlinScala[nodes.Call] = wrapped.raw
+  class AssignmentTrav(val wrapped: Steps[Assignment]) extends AnyVal {
+    private def raw: GremlinScala[Assignment] = wrapped.raw
     def target: NodeSteps[nodes.Expression] = wrapped.map(_.target)
     def source: NodeSteps[nodes.Expression] = wrapped.map(_.source)
   }
 
   class OpAstNodeExt[A <: nodes.AstNode](val node: A) extends AnyVal {
-    def inAssignment: NodeSteps[nodes.Call] =
-      node.start.inAstMinusLeaf.isCall.name(NodeTypeStarters.assignmentPattern)
+    def inAssignment: NodeSteps[Assignment] =
+      node.start.inAstMinusLeaf.isCall.name(NodeTypeStarters.assignmentPattern).map(new Assignment(_))
 
-    def assignments: NodeSteps[nodes.Call] =
-      rawTravForPattern(NodeTypeStarters.assignmentPattern)
+    def assignments: NodeSteps[Assignment] =
+      rawTravForPattern(NodeTypeStarters.assignmentPattern).map(new Assignment(_))
 
-    def arithmetics: NodeSteps[nodes.Call] =
-      rawTravForPattern(NodeTypeStarters.arithmeticPattern)
+    def arithmetics: NodeSteps[Arithmetic] =
+      rawTravForPattern(NodeTypeStarters.arithmeticPattern).map(new Arithmetic(_))
 
     private def rawTravForPattern(pattern: String): NodeSteps[nodes.Call] =
       node.ast.isCall.name(pattern)
   }
 
   class OpAstNodeTrav[A <: nodes.AstNode](val wrapped: Steps[A]) extends AnyVal {
     private def raw: GremlinScala[A] = wrapped.raw
-    def inAssignment: NodeSteps[nodes.Call] = wrapped.flatMap(_.inAssignment)
-    def assignments: NodeSteps[nodes.Call] = wrapped.flatMap(_.assignments)
-    def arithmetics: NodeSteps[nodes.Call] = wrapped.flatMap(_.arithmetics)
+    def inAssignment: NodeSteps[Assignment] = wrapped.flatMap(_.inAssignment)
+    def assignments: NodeSteps[Assignment] = wrapped.flatMap(_.assignments)
+    def arithmetics: NodeSteps[Arithmetic] = wrapped.flatMap(_.arithmetics)
```

Comment:
The hole OpAstNodeTrav class seems to be a unnecessary. Why are the three methods not defined in AstNode.scala?

Reply:
We have this class so that all steps related to AST nodes, which only make sense when the operator extension is used are in `operatorextension`. That way, it becomes possible to remove an extension without modifying `ASTNode.scala`. In addition to providing functionality, the operatorextension is used as an example of how to provide an extension. Please leave it like this.

Reply:
Ok makes sense to keep that separate. I would than suggest to not put the classes in question into the package.scala in order to keep the same layout as for the rest of the AST steps.